### DISCUSSION TITLE
Is it possible to use PRBMath V3 as a library instead of a type?

### DISCUSSION THREAD

**Discussion by partylikeits1983** [PaulRBerg/prb-math#143] [Q&A] [answered]
Hello, is it possible to use the math functions as library instead of a type? In the old version it was possible to access math functions by writing:

```
using PRBMathSD59x18 for int256;
```
Is this still possible?

**Comment by PaulRBerg** [comment, marked as answer]
Hi Alex, I'm afraid that it is not possible to use PRBMath V3 as a library. V3 is a major release (according to the [semver](https://semver.org/) definition), and so it includes breaking changes.

As per the [release notes](https://github.com/paulrberg/prb-math/releases/tag/v3.0.0) and the [announcement thread](https://twitter.com/PaulRBerg/status/1597645509094649856) on Twitter, I have deliberately refactored the library to use only [user defined value types](https://blog.soliditylang.org/2021/09/27/user-defined-value-types/) and free functions instead of the library syntax. I did this because using the `SD59x18` type instead of `int256` is much less of a footgun for end users, because it forces them to handle the fixed-point types separately (see this [comparison](https://twitter.com/PaulRBerg/status/1597645524554768388) I made between V2 and V3).

As a short term solution, what you could do is stick with the [v2.5.0](https://www.npmjs.com/package/@prb/math?activeTab=versions) release, noting that the v2.x trunk will not benefit from any further maintenance from me (except for critical bugs).

In the long term, I would encourage you to dabble into user defined value types. They really are great. I spent quite a bit of time documenting how V3 works, and in the README you can find several code snippets that should simplify the learning curve. I would imagine that for your use case, the [Convertor Functions](https://github.com/paulrberg/prb-math/tree/9ac614c890b4881caaa945115322451757bc30cd#conversion-functions) section is the most relevant.
